solr/contrib/jwt-auth/src/java/org/apache/solr/security/JWTAuthPlugin.java

```diff
@@ -104,15 +102,31 @@
   @Deprecated(since = "9.0") // Remove in 10.0
   private static final String PARAM_ALG_WHITELIST = "algWhitelist";
 
-  private static final Set<String> PROPS = ImmutableSet.of(PARAM_BLOCK_UNKNOWN,
-      PARAM_PRINCIPAL_CLAIM, PARAM_REQUIRE_EXPIRATIONTIME, PARAM_ALG_ALLOWLIST,
-      PARAM_JWK_CACHE_DURATION, PARAM_CLAIMS_MATCH, PARAM_SCOPE, PARAM_REALM, PARAM_ROLES_CLAIM,
-      PARAM_ADMINUI_SCOPE, PARAM_REDIRECT_URIS, PARAM_REQUIRE_ISSUER, PARAM_ISSUERS,
-      PARAM_TRUSTED_CERTS_FILE, PARAM_TRUSTED_CERTS,
-      // These keys are supported for now to enable PRIMARY issuer config through top-level keys
-      JWTIssuerConfig.PARAM_JWKS_URL, JWTIssuerConfig.PARAM_JWK, JWTIssuerConfig.PARAM_ISSUER,
-      JWTIssuerConfig.PARAM_CLIENT_ID, JWTIssuerConfig.PARAM_WELL_KNOWN_URL, JWTIssuerConfig.PARAM_AUDIENCE,
-      JWTIssuerConfig.PARAM_AUTHORIZATION_ENDPOINT);
+  private static final ImmutableSet<String> PROPS =
```

Comment:
Why did this type change from Set?

Reply:
Because of https://github.com/apache/solr/pull/518#discussion_r783013500
IntelliJ does NOT warn about the same, so probably not important.

Reply:
Ah, interesting. That check actually went away and became the better named and better described https://errorprone.info/bugpattern/PreferredInterfaceType

I've generally been trying to move us away from the guava types and onto jdk types using Set.of where reasonable.

Reply:
I moved this to `Set.of()` which is also immutable. Good catch.

We have more than 100 dependencies. Would be cool to make a list of them in a spreadsheet, with columns for number-of-uses, version, release-date, known CVEs, test or runtime etc. Then we could sort by fewest uses and perhaps move to JDK APIs for a lot and get rid of some weight.
We should also look at shadowed classes in Solr which are likely very old, and see if we find alternatives.